Add initial draft of a Remote Access solutions guide for TrueNAS 24.10 (#3331)
truenas/documentation: content/Solutions/RemoteAccess.md

---
title: "Accessing TrueNAS Remotely"
description: "Basic information on the variety of ways TrueNAS can be accessed remotely"
weight: 10
tags:
- gettingstarted
- vpn
- remote
- apps
---

## Overview

TrueNAS is fundamentally a storage appliance, and as such it is recommended to always be run not directly exposed to the internet.

> However, there are many instances in which a user will need to access their storage data or services remotely. In most corporate
> environments this is controlled by having access to a corporate VPN service where proper auditing and security best practices
> can be enforced. For the small business or home lab where a corporate VPN isn't available, there are many easier to deploy and
> manage services that can be used to give yourself or your clients access to your storage services and applications. This guide
> will introduce you to several of these and provide a basic walk-through of how to enable remote access to your TrueNAS.

Reviewer: Suggested rewording to make this active voice/present tense, remove contractions, and to make the text more concise.
```suggestion
However, there are many instances where a user needs to access their storage data or services remotely. 
In most corporate environments, a corporate VPN service controls internal and external access, performs proper auditing, and applies and enforces security best practices. 
For the small business or home lab, where a corporate VPN is not an option, you can choose from many easy-to-deploy and manage services that give you or your clients access to your storage services and applications. 
This guide introduces several access options and provides a basic walk-through enabling remote access to your TrueNAS.
```

> In TrueNAS 24.10 and later, there are several VPN technologies to choose from, each with their own strengths and weaknesses that
> may make them suitable for your particular environment. We will focus on three of them, all of which are available via the 
> TrueNAS App catalog for rapid deployment. 

Reviewer: Same reason for these recommended changes:
```suggestion
In TrueNAS 24.10 and later, there are several VPN technologies to choose from, each with strengths and weaknesses that make them suitable for particular environments. 
This guide focuses on three options available through the TrueNAS App catalog for rapid deployment. 
```

- Tailscale

- ZeroTier

- Wireguard

## Tailscale

Website: [Tailscale](https://www.tailscale.com/)
Pricing: Free / Paid Tier
Difficulty: Easy to Moderate

> Usability: Tailscale focuses on simplicity and ease of setup. It uses a centralized control plane and integrates with identity providers like Google or Microsoft for authentication. It’s ideal for users with limited networking knowledge.

Reviewer: ```suggestion
Usability: Tailscale focuses on simplicity and ease of setup. It uses a centralized control plane and integrates with identity providers like Google or Microsoft for authentication. It is ideal for users with limited networking knowledge.
```

> ### Installing Tailscale on TrueNAS:

Reviewer: ```suggestion
### Installing Tailscale in TrueNAS:
```

> Before you install Tailscale on TrueNAS, you will need to create an [Auth Key](https://login.tailscale.com/admin/settings/keys) via your Tailscale account. [More Information](https://tailscale.com/kb/1085/auth-keys)

Reviewer: ```suggestion
Before installing Tailscale in TrueNAS, create an [Auth Key](https://login.tailscale.com/admin/settings/keys) through your Tailscale account. [More Information](https://tailscale.com/kb/1085/auth-keys)
```

> Once your Auth Key has been created, you can navigate to the TrueNAS Apps Catalog and install Tailscale:

Reviewer: ```suggestion
After creating your Auth Key you can navigate to the TrueNAS Apps Catalog and install Tailscale:
```

> Apps -> Discover Apps -> (Search for Tailscale) -> Install

Reviewer: ```suggestion
**Apps -> Discover Apps ->** (Search for Tailscale) **-> Install**
```

> On the Installation page, scroll down to the `AuthKey` field, paste in the key you created previously. Leave all other options at their defaults, then scroll down and click `Install`.

Reviewer: Fixed emphasis, changing it from code to bold, and suggested rewording to make the text concise.
```suggestion
On the Installation page, scroll down to the **AuthKey** field, and paste the previously created key. Accept all other default settings, and then scroll down and click **Install**.
```

> Once the installation completes and you see the App status as "Running" you can navigate back to your [Tailscale Machines](https://login.tailscale.com/admin/machines) list and you will see your TrueNAS system listed.

Reviewer: We don't say "once", we use after or when based on how something is worded.  Suggestions to make active voice/present tense and other text changes
```suggestion
After the installation completes and you see the app status as **Running** you can navigate back to your [Tailscale Machines](https://login.tailscale.com/admin/machines) list to see your TrueNAS system listed.
```

> Using the `Addresses` listed for TrueNAS, you can now reach your TrueNAS host, including Shares, Apps and other services from your other [Tailscale clients](https://tailscale.com/download) anywhere in the world. 

Reviewer: ```suggestion
Using the **Addresses**  listed for TrueNAS, you can now reach your TrueNAS host, including shares, apps, and other services from your other [Tailscale clients](https://tailscale.com/download) anywhere in the world. 
```

## ZeroTier

Website: [ZeroTier](https://www.zerotier.com/)
Pricing: Free / Paid Tier
Difficulty: Moderate to Hard

> Usability: While still being user-friendly, ZeroTier offers more advanced configuration options, such as managing multiple networks, IP ranges, and multicast settings. This makes it better suited for users with technical expertise.

Reviewer: ```suggestion
Usability: User-friendly ZeroTier offers more advanced configuration options, such as managing multiple networks, IP ranges, and multicast settings. This makes it better suited for users with technical expertise.
```

### Installing ZeroTier on TrueNAS:

> Before you install ZeroTier on TrueNAS, you will need to create an [Network](https://my.zerotier.com/network) via your ZeroTier account. [More Information](https://docs.zerotier.com/start)

Reviewer: ```suggestion
Before installing ZeroTier on TrueNAS, create a [Network](https://my.zerotier.com/network) through your ZeroTier account. [More Information](https://docs.zerotier.com/start)
```

> Once your Network has been created, copy the `Network ID` and then you can navigate to the TrueNAS Apps Catalog to install ZeroTier:

Reviewer: ```suggestion
After creating the Network, copy the **Network ID** and then navigate to the TrueNAS Apps Catalog to install ZeroTier:
```

> Apps -> Discover Apps -> (Search for ZeroTier) -> Install

Reviewer: ```suggestion
**Apps -> Discover Apps ->** (Search for ZeroTier) **-> Install**
```

> On the Installation page, scroll down to the `ZeroTier Configuration -> Networks` field. Click to add a network and paste in the network key you copied previously. Leave all other options at their defaults, then scroll down and click `Install`.

Reviewer: ```suggestion
On the installation screen, scroll down to the **ZeroTier Configuration -> Networks** field.
Click to add a network and paste in the network key you copied previously. Accept all other default settings, and then scroll down to click **Install**.
```

> Once the installation completes and you see the App status as "Running" you can navigate back to your [ZeroTier Networks](https://my.zerotier.com/networks) list, click your network and you will see your TrueNAS system waiting for approval.

Reviewer: ```suggestion
After the installation completes and the App status changes to "Running", you can navigate back to your [ZeroTier Networks](https://my.zerotier.com/networks) list, and click on your network to see your TrueNAS system waiting for approval.
```

> Click to authorize TrueNAS and you will have successfully finished joined your TrueNAS system to a ZeroTier network.

Reviewer: ```suggestion
Click to authorize TrueNAS and you have successfully finished and joined your TrueNAS system to a ZeroTier network.
```

> Using the `Addresses` listed for TrueNAS, you can now reach your TrueNAS host, including Shares, Apps and other services from your other [ZeroTier clients](https://www.zerotier.com/download/) anywhere in the world. 

Reviewer: ```suggestion
Using the **Addresses** listed for TrueNAS, you can reach your TrueNAS host, including shares, apps, and other services from your other [ZeroTier clients](https://www.zerotier.com/download/) anywhere in the world. 
```

## Wireguard

Website: [WireGuard](https://www.wireguard.com)
Pricing: Free
Difficulty: Hard
Usability: Wireguard requires manual setup and configuration. You need to manage keys, IPs, and routing yourself.

### Installing Wireguard on TrueNAS:

> Wireguard can be setup and configured by installing the `WG Easy` Application in the TrueNAS app catalog. [Click Here](https://www.truenas.com/docs/truenasapps/stableapps/installwgeasyapp/) for more detailed information on how to setup and configure.

Reviewer: ```suggestion
Wireguard can be setup and configured by installing the **WG Easy** application in the TrueNAS app catalog. [Click here](https://www.truenas.com/docs/truenasapps/stableapps/installwgeasyapp/) for more detailed information on  setting up and configuring the app.
```
